src/pages/index.js

```diff
@@ -90,44 +102,88 @@ const profileEditPopup = new PopupWithForm(
 profileEditPopup.setEventListeners();
 
 // Profile Edit Popup Button
+const profileEditButton = document.querySelector("#profile-edit-button");
+
 profileEditButton.addEventListener("click", () => {
-  const userInfoData = userInfo.getUserInfo();
+  if (userInfoInstance) {
+    const userInfoData = userInfoInstance.getUserInfo();
 
-  profileTitleInput.value = userInfoData.name;
-  profileDescriptionInput.value = userInfoData.job;
+    profileTitleInput.value = userInfoData.name;
+    profileDescriptionInput.value = userInfoData.job;
 
-  profileEditPopup.open();
+    profileFormValidator.resetValidation();
+    profileEditPopup.open();
+  }
 });
 
-// Profile Changes Save
 function handleProfileSaveSubmit(inputValues) {
-  userInfo.setUserInfo({
+  const saveButton = document.getElementById("profile-modal-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  userInfoInstance.setUserInfo({
     name: inputValues.title,
     job: inputValues.description,
+    avatar: avatarImg.src,
   });
+
+  const newInfo = {
+    name: inputValues.title,
+    about: inputValues.description,
+    avatar: avatarImg.src,
+  };
+
+  apiUpdateProfileInfo(newInfo).then(() => (saveButton.textContent = "Save"));
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

**Please, correct it everywhere**